agent/xds/listeners.go

```diff
@@ -1436,6 +1508,13 @@ func makeSNIClusterFilter() (*envoy_listener_v3.Filter, error) {
 	return makeFilter("envoy.filters.network.sni_cluster", &envoy_sni_cluster_v3.SniCluster{})
 }
 
+func makeSNIDynamicForwardProxyFilter() (*envoy_listener_v3.Filter, error) {
+	return makeFilter("envoy.filters.network.sni_dynamic_forward_proxy", &envoy_sni_dynamic_forward_proxy_v3.FilterConfig{
+		DnsCacheConfig: getCommonDNSCacheConfiguration(),
+		PortSpecifier:  &envoy_sni_dynamic_forward_proxy_v3.FilterConfig_PortValue{PortValue: 443},
```

Comment:
Why the port is hardcoded to 443 here?

Reply:
:shame: I just missed it when copying over the configuration. I'll make sure this is the Destination Port.